fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -458,6 +475,414 @@ Value makePadding(int size) {
 	return pad.substr(0, size);
 }
 
+struct IRangeFileWriter {
+public:
+	virtual Future<Void> padEnd(bool final) = 0;
+
+	virtual Future<Void> writeKV(Key k, Value v) = 0;
+
+	virtual Future<Void> writeKey(Key k) = 0;
+
+	virtual Future<Void> finish() = 0;
+
+	virtual ~IRangeFileWriter() {}
+};
+
+struct SnapshotFileBackupEncryptionKeys {
+	Reference<BlobCipherKey> textCipherKey;
+	Reference<BlobCipherKey> headerCipherKey;
+	StringRef ivRef;
+};
+
+// File Format handlers.
+// Both Range and Log formats are designed to be readable starting at any 1MB boundary
```

Comment:
nit: opportunity to update the comment `1 MB` boundary is _not_ hardcoded I think?